### PR TITLE
Reduce some direct package:archive usage

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Use os.unzip for inteliij, which has the benefit of using the host machine unzip where available. Adds a note about potentially using powershell exposed zip/unzip on windows instead.

Removes dead code from web asset server. This code isn't needed since we don't use build_runner for dart2js anymore.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Back to WIP

File: packages/flutter_tools/lib/src/intellij/intellij.dart [hunk 1]
```diff
@@ -56,18 +55,25 @@ class IntelliJPlugins {
...
       final int start = content.indexOf(versionStartTag);
       final int end = content.indexOf('</version>', start);
+      try {
+        tempDirectory?.deleteSync(recursive: true);
+      } on Exception catch (_) {
+        printTrace('Failed to delete temp directory: ${tempDirectory?.path}');
+      }
       return content.substring(start + versionStartTag.length, end);
-    } catch (_) {
+    } on Exception catch (_) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/intellij/intellij.dart]: This was catching a type error in my test

File: packages/flutter_tools/test/general.shard/intellij/intellij_test.dart [hunk 2]
```diff
@@ -1,108 +1,97 @@
...
+
+''';
+const String flutterPluginContents = r'''
+<idea-plugin version="2">
+<name>Flutter</name>
+<version>0.1.3</version>
+</idea-plugin>
+''';
+
 void main() {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/intellij/intellij_test.dart]: Most of the delta here is removing two nested groups

File: packages/flutter_tools/lib/src/base/os.dart [hunk 3]
```diff
@@ -2,6 +2,11 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// package:archive is used to provide zip/unzip only on Windows host machines. Since
+// Windows 8 and powershell 3, the native zip/unzip functionality has been exposed
+// and should be preferred over the dart implementation.
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/base/os.dart]: Should we update the APIs herein to make that determination?

File: packages/flutter_tools/lib/src/intellij/intellij.dart [hunk 4]
```diff
@@ -56,18 +55,25 @@ class IntelliJPlugins {
...
+      os.unzip(fs.file(jarPath), tempDirectory);
+      final File file = tempDirectory
+        .childDirectory('META-INF')
+        .childFile('plugin.xml');
+      final String content = file.readAsStringSync();
       const String versionStartTag = '<version>';
       final int start = content.indexOf(versionStartTag);
       final int end = content.indexOf('</version>', start);
+      try {
+        tempDirectory?.deleteSync(recursive: true);
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/intellij/intellij.dart]: tempDirectory can't be null here, right?

File: packages/flutter_tools/lib/src/base/os.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/base/os.dart]: This should probably be captured in a separate bug and not a comment.

File: packages/flutter_tools/lib/src/intellij/intellij.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/intellij/intellij.dart]: Fixed

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_tools/lib/src/intellij/intellij.dart (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/lib/src/intellij/intellij.dart]: What would this (and the one on line 72 as well) be other than `FileSystemException`? Should this one `printError()`?

**Pull request merged**